src/ebmc/ebmc_properties.h

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************\
+
+Module: Data Structure for the Properties
+
+Author: Daniel Kroening, dkr@amazon.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_EBMC_PROPERTIES_H
+#define CPROVER_EBMC_PROPERTIES_H
+
+#include <util/cmdline.h>
+#include <util/message.h>
+
+#include <solvers/prop/literal.h>
+#include <trans-netlist/trans_trace.h>
+
+#include "transition_system.h"
+
+class ebmc_propertiest
+{
+public:
+  struct propertyt
+  {
+  public:
+    unsigned number;
+    irep_idt name;
+    source_locationt location;
+    std::string expr_string;
+    irep_idt mode;
+    exprt expr;
+    bvt timeframe_literals;             // bit level
+    exprt::operandst timeframe_handles; // word level
+    std::string description;
+    enum class statust
+    {
+      DISABLED,
+      SUCCESS,
+      FAILURE,
+      UNKNOWN
+    } status;
```

Comment:
With `= 0` above and `status = UNKNOWN` it should be possible to make the default constructor `= default;`